src/main/java/org/spongepowered/api/item/inventory/Container.java

```diff
@@ -24,60 +24,50 @@
-
-import java.util.Set;
+import java.util.List;
+import java.util.Optional;
 
 /**
  * A Container is effectively a <em>ViewModel</em> for a particular set of
  * {@link Inventory} objects used to allow players to interact
  * with the Inventories, usually via a GUI (the View).
  */
-public interface Container extends Interactable {
+public interface Container extends Inventory {
 
-    /**
-     * Gets the current viewers looking at this Inventory.
+     /**
+     * Returns whether given slot is part of the viewed inventories
+     * but not part of the viewers own inventory.
      *
-     * @return The current viewers of this inventory
-     */
-    Set<Player> getViewers();
-
-    /**
-     * Checks for whether this Inventory currently has viewers.
+     * <p>Examples for viewed inventory slots:</p>
+     * <p>All the slots of the opened chest and not the player grid below.</p>
+     * <p>The slots of the crafting grid in the opened player inventory.</p>
      *
-     * @return True if viewers are currently looking at this inventory
+     * @param slot The slot to check.
+     * @return {@code true} when the slot is part of the viewed inventories.
      */
-    boolean hasViewers();
+    boolean isViewedSlot(Slot slot);
 
     /**
-     * Shows this Inventory to the given viewer.
+     * Returns the list of viewed inventories.
+     * <p>This is usually at least the inventory a player opened and the players inventory.</p>
+     * <p>It is not necessary, that all slots of the viewed inventories are visible or interactable with.</p>
      *
-     * @param viewer The viewer to show this inventory to
-     * @throws IllegalArgumentException if a {@link PluginContainer} is not the root of the cause
+     * @return the list of viewed inventories.
      */
-    void open(Player viewer) throws IllegalArgumentException;
+    List<Inventory> getViewed();
 
     /**
-     * Stops showing this Inventory to the given viewer.
+     * Sets the viewing players cursor item.
      *
-     * @param viewer The viewer to stop showing this inventory to
-     * @throws IllegalArgumentException if a {@link PluginContainer} is not the root of the cause
+     * @param item The item to set.
      */
-    void close(Player viewer) throws IllegalArgumentException;
+    void setCursor(ItemStack item);
 
     /**
-     * Returns whether given slot is part of the viewed inventories
-     * but not part of the viewers own inventory.
+     * Gets the viewing players cursor item.
      *
-     * <p>Examples for viewed inventory slots:</p>
-     * <p>All the slots of the opened chest and not the player grid below.</p>
-     * <p>The slots of the crafting grid in the opened player inventory.</p>
-     *
-     * @param slot The slot to check.
-     * @return {@code true} when the slot is part of the viewed inventories.
+     * @return The players cursor item.
      */
-    boolean isViewedSlot(Slot slot);
+    Optional<ItemStack> getCursor();
```

Comment:
Should we consider empty stacks for this?

Reply:
Its actually Optional  for the case you have a Container which has been closed already.
If no player is looking at it there is no cursor.